### DISCUSSION TITLE
Using TSDB analyze on the head-block

### DISCUSSION THREAD

**Discussion by amityahav** [prometheus/prometheus#11874] [Q&A]
Hey, is there a way to perform the promtool tsbd analyze on the in-memory head-block? my use case is trying to get the highest cardinality label names of the latest time, or alternatively does the 'tsdb/status/' endpoint returns this latest-time statistics?

Thanks

**Comment by amityahav** [comment]
just went over the code and it seems that the endpoint retreives its stats from the Head

**Comment by ringerc** [comment]
You can take a TSDB snapshot via the Prometheus admin API with e.g. `curl -XPOST :9090/api/v1/admin/tsdb/snapshot` (adjust URI as needed).

Then you can run `promtool tsdb analyze` or other promtool commands on the exported snapshot block under `data/snapshots`. To do this, you must create an empty `wal` directory in `data/snapshots`, because these tools assume there's a prometheus WAL in the same level of the directory as the data block, but there won't be for a snapshot.

See also https://www.robustperception.io/taking-snapshots-of-prometheus-data/
